README_python.md

```diff
@@ -1,17 +1,46 @@
 # Halide Bindings for Python
 
+<!-- MarkdownTOC autolink="true" -->
+
+- [Python Requirements](#python-requirements)
+- [Compilation Instructions](#compilation-instructions)
+- [Documentation and Examples](#documentation-and-examples)
+- [Differences from C++ API](#differences-from-c-api)
+- [Example of Simple Usage](#example-of-simple-usage)
+- [Halide Generators In Python](#halide-generators-in-python)
+    - [Writing a Generator in Python](#writing-a-generator-in-python)
+        - [@hl.generator\("name"\)](#hlgeneratorname)
+        - [hl.GeneratorParam](#hlgeneratorparam)
+        - [hl.InputBuffer, hl.InputScalar](#hlinputbuffer-hlinputscalar)
+        - [hl.OutputBuffer, hl.OutputScalar](#hloutputbuffer-hloutputscalar)
+        - [Names](#names)
+        - [generate\(\) method](#generate-method)
+    - [Using a Generator for JIT compilation](#using-a-generator-for-jit-compilation)
+    - [Using a Generator for AOT compilation](#using-a-generator-for-aot-compilation)
+    - [Calling Generator-Produced code from Python](#calling-generator-produced-code-from-python)
+    - [Advanced Generator-Related Topics](#advanced-generator-related-topics)
+        - [Generator Aliases](#generator-aliases)
+        - [Dynamic Inputs and Outputs](#dynamic-inputs-and-outputs)
+        - [Calling a Generator Directly](#calling-a-generator-directly)
+        - [The Lifecycle Of A Generator](#the-lifecycle-of-a-generator)
+        - [Notable Differences Between C++ and Python Generators](#notable-differences-between-c-and-python-generators)
+- [Keeping Up To Date](#keeping-up-to-date)
+- [License](#license)
+
+<!-- /MarkdownTOC -->
+
 Halide provides Python bindings for most of its public API. Only Python 3.x is
 supported; at this time, Python 3.8 (or higher) is recommended. The Python
 bindings are supported on 64-bit Linux, OSX, and Windows systems.
 
 In addition to the ability to write just-in-time Halide code using Python, you
-can also use existing Halide Generators (written in C++) to produce Python
-extensions that can be used within Python code.
+can write Generators using the Python bindings, which can simplify build-system
```

Comment:
Make a link from Generators to the appropriate section of this document.

Reply:
Done